pycbc/waveform/waveform_modes.py

```diff
@@ -0,0 +1,491 @@
+_mode_waveform_fd = {#'IMRPhenomXHM': get_imrphenomhm_modes,
+                     #'IMRPhenomXPHM' : get_imrphenomhm_modes,
+                    }
+
+
+def fd_waveform_mode_approximants():
+    """Frequency domain approximants that will return separate modes."""
+    return sorted(_mode_waveform_fd.keys())
+
+
+def td_waveform_mode_approximants():
+    """Time domain approximants that will return separate modes."""
+    return sorted(_mode_waveform_td.keys())
+
+
+def get_fd_waveform_modes(template=None, **kwargs):
+    r"""Generates frequency domain waveforms, but does not sum over the modes.
+
+    The returned values are the frequency-domain equivalents of the real and
+    imaginary parts of the complex :math:`\mathfrak{{h}}_{{\ell m}}(t)` time
+    series. In other words, the returned values are equivalent to the Fourier
+    Transform of the two time series returned by
+    :py:func:`get_td_waveform_modes`; see that function for more details.
+
+    Parameters
+    ----------
+    template: object
+        An object that has attached properties. This can be used to subsitute
+        for keyword arguments.
+    {params}
+
+    Returns
+    -------
+    ulm : dict
+        Dictionary of mode tuples -> fourier transform of the real part of the
+        hlm time series, as a :py:class:`pycbc.types.FrequencySeries`.
+    vlm : dict
+        Dictionary of mode tuples -> fourier transform of the imaginary part of
+        the hlm time series, as a :py:class:`pycbc.types.FrequencySeries`.
+    """
+    params = props(template, **kwargs)
+    required = parameters.fd_required
+    check_args(params, required)
+    apprx = params['approximant']
+    if apprx not in _mode_waveform_fd:
+        raise ValueError("I don't support approximant {}, sorry"
+                         .format(apprx))
+    return _mode_waveform_fd[apprx](**params)
+
+
+get_fd_waveform_modes.__doc__ = _formatdocstrlist(
+    get_fd_waveform_modes.__doc__, parameters.fd_waveform_params,
+    skip_params=['inclination', 'coa_phase'])
+
+
+def get_td_waveform_modes(template=None, **kwargs):
+    r"""Generates time domain waveforms, but does not sum over the modes.
+
+    The returned values are the real and imaginary parts of the complex
+    :math:`\mathfrak{{h}}_{{\ell m}}(t)`. These are defined such that the plus
+    and cross polarizations :math:`h_{{+,\times}}` are:
+
+    .. math::
+
+       h_{{+,\times}}(\theta, \phi; t) = (\Re, -\Im) \sum_{{\ell m}}
+        {{}}_{{-2}}Y_{{\ell m}}(\theta, \phi) \mathfrak{{h}}_{{\ell m}}(t).
+
+
+    Parameters
+    ----------
+    template: object
+        An object that has attached properties. This can be used to subsitute
+        for keyword arguments.
+    {params}
+
+    Returns
+    -------
+    ulm : dict
+        Dictionary of mode tuples -> real part of the hlm, as a
+        :py:class:`pycbc.types.TimeSeries`.
+    vlm : dict
+        Dictionary of mode tuples -> real part of the hlm, as a
```

Comment:
`real` -> `imaginary` in the docstring, I think.

Reply:
Whoops! Thanks, fixed.